pkgs/os-specific/linux/ithc/default.nix

```diff
@@ -0,0 +1,38 @@
+{ lib, stdenv, fetchFromGitHub, kernel }:
+
+stdenv.mkDerivation rec {
+  pname = "ithc";
+  version = "unstable-2022-06-07";
+
+  src = fetchFromGitHub {
+    owner = "quo";
+    repo = "ithc-linux";
+    rev = "5af2a2213d2f3d944b19ec7ccdb96f16d56adddb";
+    hash = "sha256-p4TooWUOWPfNdePE18ESmRJezPDAl9nLb55LQtkJiSg=";
+  };
+
+  setSourceRoot = ''
+    export sourceRoot=$(pwd)/source/src
+  '';
```

Comment:
```suggestion
  sourceRoot = "source/src";
```

Reply:
I've just pushed an update.
The original version was based on the `xpadneo` module who was defined this way.
When I applied your suggestion it failed to build so I made additional modifications (especially removed the call to depmod). Can you confirm this is the right way to build a module on nixos ?

Reply:
Doesn't look that wrong to me. Did you try rebuilding on your system if it works?

Reply:
I tried to apply the suggestion and it ended up with the following log:
```
$> nix-env -f .  -iA linuxKernel.packages.linux_5_15.ithc
replacing old 'ithc-unstable-2022-06-07'
installing 'ithc-unstable-2022-06-07'
this derivation will be built:
  /nix/store/fpxky28f4aia9wlrihacch8bil4r1hzf-ithc-unstable-2022-06-07.drv
building '/nix/store/fpxky28f4aia9wlrihacch8bil4r1hzf-ithc-unstable-2022-06-07.drv'...
unpacking sources
unpacking source archive /nix/store/6n7p7c6z9cfrxja0fgninn277rlspcr5-source
source root is source/src
patching sources
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/qm2lv1gpbyn0rsfai40cbvj3h4gz69yc-bash-5.1-p16/bin/bash O=\$\(buildRoot\) CC=/nix/store/6mx62f4mdv5y1awi7y2mp23a76gg0ba7-gcc-wrapper-11.3.0/bin/cc HOSTCC=/nix/store/6mx62f4mdv5y1awi7y2mp23a76gg0ba7-gcc-wrapper-11.3.0/bin/cc ARCH=x86_64 -C /nix/store/fxq9d28smhfh5n7fvi2mggghcjx3b63r-linux-5.15.56-dev/lib/modules/5.15.56/build M=\$\(sourceRoot\) VERSION=unstable-2022-06-07 modules
make: Entering directory '/nix/store/fxq9d28smhfh5n7fvi2mggghcjx3b63r-linux-5.15.56-dev/lib/modules/5.15.56/build'
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
mkdir: cannot create directory 'source/src': Permission denied
/nix/store/fxq9d28smhfh5n7fvi2mggghcjx3b63r-linux-5.15.56-dev/lib/modules/5.15.56/source/scripts/Makefile.build:44: /nix/store/fxq9d28smhfh5n7fvi2mggghcjx3b63r-linux-5.15.56-dev/lib/modules/5.15.56/source/source/src/Makefile: No such file or directory
make[1]: *** No rule to make target '/nix/store/fxq9d28smhfh5n7fvi2mggghcjx3b63r-linux-5.15.56-dev/lib/modules/5.15.56/source/source/src/Makefile'.  Stop.
make: *** [/nix/store/fxq9d28smhfh5n7fvi2mggghcjx3b63r-linux-5.15.56-dev/lib/modules/5.15.56/source/Makefile:1886: source/src] Error 2
make: Leaving directory '/nix/store/fxq9d28smhfh5n7fvi2mggghcjx3b63r-linux-5.15.56-dev/lib/modules/5.15.56/build'
error: builder for '/nix/store/fpxky28f4aia9wlrihacch8bil4r1hzf-ithc-unstable-2022-06-07.drv' failed with exit code 2;
       last 10 log lines:
       > mkdir: cannot create directory 'source/src': Permission denied
       > mkdir: cannot create directory 'source/src': Permission denied
       > mkdir: cannot create directory 'source/src': Permission denied
       > mkdir: cannot create directory 'source/src': Permission denied
       > mkdir: cannot create directory 'source/src': Permission denied
       > mkdir: cannot create directory 'source/src': Permission denied
       > /nix/store/fxq9d28smhfh5n7fvi2mggghcjx3b63r-linux-5.15.56-dev/lib/modules/5.15.56/source/scripts/Makefile.build:44: /nix/store/fxq9d28smhfh5n7fvi2mggghcjx3b63r-linux-5.15.56-dev/lib/modules/5.15.56/source/source/src/Makefile: No such file or directory
       > make[1]: *** No rule to make target '/nix/store/fxq9d28smhfh5n7fvi2mggghcjx3b63r-linux-5.15.56-dev/lib/modules/5.15.56/source/source/src/Makefile'.  Stop.
       > make: *** [/nix/store/fxq9d28smhfh5n7fvi2mggghcjx3b63r-linux-5.15.56-dev/lib/modules/5.15.56/source/Makefile:1886: source/src] Error 2
       > make: Leaving directory '/nix/store/fxq9d28smhfh5n7fvi2mggghcjx3b63r-linux-5.15.56-dev/lib/modules/5.15.56/build'
       For full logs, run 'nix log /nix/store/fpxky28f4aia9wlrihacch8bil4r1hzf-ithc-unstable-2022-06-07.drv'.
```

I deduced that the files were not created in the correct location.